zetasql/parser/bison_parser.y

```diff
@@ -3375,27 +3382,45 @@ export_model_statement:
     ;
 
 grant_statement:
-    "GRANT" privileges "ON" identifier path_expression "TO" grantee_list
+    "GRANT" privileges "ON" grant_object_kind grant_path "TO" grantee_list opt_grant_option
```

Comment:
```yacc
grant_statement:
  "GRANT" privileges "ON" identifier path_expression "TO" grantee_list
   {}
  | "GRANT" privileges "ON" identifier "*" "TO" grantee_list
  {}
  | "GRANT" privileges "ON" path_expression "TO" grantee_list
  {}
  | "GRANT" privileges "ON" "*" "TO" grantee_list
  {}
  ;
```

same for revoke statement

Reply:
This solution seems more complicated than using grant path

Reply:
and according to previous tests, it is easy to conflict with the subsequent keyword  `TO`

Reply:
just need a bit code in bison:

like
```
"GRANT" privileges "ON" identifier "*" "TO" grantee_list
  {
    auto* identifier = parser->MakeIdentifier(@5, parser->GetInputText(@5));                                                                                                                                                     
    auto* path_expression = MAKE_NODE(ASTPathExpression, @5, {identifier});                                                                                                                                                      
    $$ = MAKE_NODE(ASTGrantStatement, @$, {$2, $4, path_expression, $7});
  }
```

I think `ASTPathExpression` is good enough to handle extra `*` keyword

Reply:
ok，let me try

Reply:
I think '`*.t1` does not make sense, not a good pratice to select table t1 in any db. 
I see mysql have syntax 
```
priv_level: {
    *
  | *.*
  | db_name.*
  | db_name.tbl_name
  | tbl_name
  | db_name.routine_name
}
```

The only thing not handled is `*.*`, and it can represented by extra rule:
```
"GRANT" privileges "ON" "*" ".*" "TO" grantee_list
```
Not really sure but do `*` and `*.*` have a real big difference ?

Reply:
There are still some problems with enumerating all situations in the above way. so i convert the grant path to path expression finally. it can be reconstruct later if you want.